compiler/src/dotty/tools/dotc/typer/Inliner.scala

```diff
@@ -843,7 +924,16 @@ class Inliner(call: tpd.Tree, rhsToInline: tpd.Tree)(using Context) {
 
     // Apply inliner to `rhsToInline`, split off any implicit bindings from result, and
     // make them part of `bindingsBuf`. The expansion is then the tree that remains.
-    val expansion = inliner.transform(rhsToInline)
+    val expansion0 = inliner.transform(rhsToInline)
+    val expansion =
+      if opaqueProxies.nonEmpty && !inlinedMethod.is(Transparent) then
+        expansion0.cast(call.tpe)(using ctx.withSource(expansion0.source))
+          // the cast makes sure that the sealing with the declared type
+          // is type correct. Without it we might get problems since the
+          // expression's type is the opaque alias but the call's type is
+          // the opaque type itself. An example is in pos/opaque-inline1.scala.
```

Comment:
This branch is never actually reached for `pos/opaque-inline1.scala`, it is reached for `pos/opaque-inline.scala` but at that point `expansion0.tpe` and `call.tpe` are both `refined.Positive` so the cast doesn't do anything (and the test passes if I delete it).